File: frontend/movie-app/src/app/movie.ts
export class Movie {
    id: number;
    title: string;
    overview: string;
    backdrop_path: string;
    vote_average: number;
    recommended: boolean;

    constructor(id: number, title: string, overview: string, backdrop_path: string, vote_average: number, recommended: boolean) {
        this.id = id;
        this.title = title;
        this.overview = overview;
        this.backdrop_path = backdrop_path;
        this.vote_average = vote_average;
        this.recommended = recommended;
    }
}

File: server/Models/Movie.cs
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

public partial class Movie
{
    [DatabaseGenerated(DatabaseGeneratedOption.None)]
    [JsonProperty("id")]
    public long Id { get; set; }

    [JsonProperty("title")]
    public string Title { get; set; }

    [JsonProperty("overview")]
    public string Overview { get; set; }

    [JsonProperty("backdrop_path")]
    public string BackdropPath { get; set; }

    [JsonProperty("vote_average")]
    public double VoteAverage { get; set; }

    [NotMapped]
    [JsonProperty("recommended")]
    public bool IsRecommended { get; set; }
}